### PR TITLE
Callback refactor Initialize and CreateWallet

### PR DESCRIPTION
**Pull request by jdkuki**
Resolves brave/brave-browser#5135
Resolves brave/brave-browser#5134

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/brave-ios/Ledger/BATBraveLedger.mm [hunk 1]
```diff
@@ -110,7 +109,9 @@ - (instancetype)initWithStateStoragePath:(NSString *)path
 
     ledgerClient = new NativeLedgerClient(self);
     ledger = ledger::Ledger::CreateInstance(ledgerClient);
-    ledger->Initialize();
+    ledger->Initialize(^(int32_t result){
```
Comment (jdkuki) [inline vendor/brave-ios/Ledger/BATBraveLedger.mm]: @kylehickinson In `RewardsService`, we're using the same callback as `createWallet`. Should IOS behave the same?

Comment (kylehickinson) [reply vendor/brave-ios/Ledger/BATBraveLedger.mm]: Not sure what you mean? Like the contents of the callback? Or just the same type of callback (`int32_t result`)

Comment (jdkuki) [reply vendor/brave-ios/Ledger/BATBraveLedger.mm]: The contents of the callback, ex `RewardsServiceImpl` uses `OnWalletInitialized` for both `Initialize` and `CreateWallet`

Comment (kylehickinson) [reply vendor/brave-ios/Ledger/BATBraveLedger.mm]: Ah I see what you mean. Basically that we want to call the wallet initialized observers. Currently there's more logic involved around creating wallets callback than just observers, so how about for now lets just makes `Initalize`'s callback do this:

```
ledger->Initialize(^(int32_t result){
  for (BATBraveLedgerObserver *observer in self.observers) {
    if (observer.walletInitalized) {
      observer.walletInitalized(static_cast<BATResult>(result));
    }
  }
});
```

And I will cleanup the implementation at a later date

Comment (jdkuki) [reply vendor/brave-ios/Ledger/BATBraveLedger.mm]: Will do, thanks for the feedback.

**Pull request merged**